herckules/src/main/java/edu/asu/diging/lerna/herckules/db/impl/ProjectDBConnector.java

```diff
@@ -0,0 +1,122 @@
+package edu.asu.diging.lerna.herckules.db.impl;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.TypedQuery;
+
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.diging.lerna.herckules.authentication.impl.IExceptionMessages;
+import edu.asu.diging.lerna.herckules.db.IProjectDBConnector;
+import edu.asu.diging.lerna.herckules.domain.impl.Project;
+import edu.asu.diging.lerna.herkules.exception.*;
+
+/**
+ * This class implements {@link IProjectDBConnector} functions. This component
+ * layer class is used to retrieve, or delete a project from a objectDB
+ * database. It also has functions for adding or modifying a project.
+ * 
+ * @author Tamalika Mukherjee
+ * @author Vineel Vutukuri
+ */
+
+@Component
+@Transactional
+public class ProjectDBConnector implements IProjectDBConnector {
+
+	@PersistenceContext
+	private EntityManager manager;
+
+	/**
+	 * Retrieves the project from the database.
+	 * 
+	 * @param Project
+	 *            The project object that is to be retrieved to the database.
+	 * @return true Returns true if the project is successfully retrieved from
+	 *         the database.
+	 */
+	@Transactional
+	public Project retrieveProject(Project project) {
+
+		TypedQuery<Project> query = manager
+				.createQuery(
+						"SELECT project FROM Project project WHERE project.creator == :creator",
+						Project.class);

```

Comment:
This doesn't make much sense. If I understand that correctly, you querying for all projects with creator, creator. But then you return only the first one you find. You would either want to find the project that you specify in your parameter or return all of the projects of the user.